crates/base/src/datetimes.rs

```diff
@@ -131,28 +152,60 @@ impl TimeZoneId {
         TZ_VARIANTS.iter().find_map(|tz| {
             let some_time = tz.ymd(1, 1, 1).and_hms(0, 0, 0);
             if some_time.offset().fix() == ctz {
-                return Some(TimeZoneId(*tz as u16));
+                return Some(TimeZoneId(*tz as i16));
             }
             None
         })
     }
 
-    // SAFETY: the timezone id only comes from a variant of `chrono_tz::Tz`, so it
-    // is always valid
+    /// Return the corresponding `chrono_tz::Tz` variant for the non-negative values.
+    ///
+    /// SAFETY: for non-negative values, the timezone id only comes from a variant of
+    /// `chrono_tz::Tz`, so it is always valid.
     fn tz(self) -> Tz {
+        debug_assert!(self.0 >= 0);
         unsafe { std::mem::transmute(self.0) }
     }
 
+    /// Return the name of timezone.
+    ///
+    /// * For negative values, its lower byte `i8` means the offset in quarters, will return the
+    ///   format as `"+XX:XX"`;
+    /// * For non-negative values, its lower byte 'i8' means a variant of `chrono_tz::Tz`, will
+    ///   return its name.
     pub fn name(self) -> &'static str {
-        self.tz().name()
+        if self.0 < 0 {
+            static NAMES: SyncLazy<Vec<String>> = SyncLazy::new(|| {
+                (-24 * 4 + 1..24 * 4)
```

Comment:
just use offset in sec to discard such complex computing

Reply:
The name's calculation is still needed here. it is a little bit complicated but will be calculated only once.